cluster-autoscaler/simulator/dynamicresources/snapshot/snapshot.go

```diff
@@ -0,0 +1,48 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package snapshot
+
+import (
+	schedulerframework "k8s.io/kubernetes/pkg/scheduler/framework"
+)
+
+// Snapshot contains a snapshot of all DRA objects taken at a ~single point in time.
+type Snapshot struct {
```

Comment:
I have some reservations to the naming: We will have ClusterSnapshot that has its ClusterSnapshotStore, which in turn has its drasnapshot.Snapshot. Do we want to use Snapshot here? Maybe SnapshotStore would be better, to make it consistent with our agreed indication of storing data? Maybe even full DraSnapshotStore, although this name is ugly and I don't like it (it's probably also not consistent with best practises).

Reply:
Hmm, on the other hand in the context of the `dynamicresources` package the name `Snapshot` seems the most accurate to me (since we don't have the 2 layers or any interfaces here - why "Store"?), and it's very simple. So I guess it's a tradeoff between readability in `clustersnapshot` and `dynamicresources`? I'm leaning more towards keeping the simple `dynamicresources.Snapshot` name, but it's not a strong opinion. @jackfrancis WDYT?

Reply:
Just to note, I don't have a very hard preference here, I am reasonably okay with merging it as is.

Reply:
In that case I'd prefer to keep the current name, if only to reduce the rebasing because of time constraints.

Reply:
I'd lean towards `Snapshot` as well (w/ the namespacing of the `dynamicresources` package providing some context to disambiguate between other snapshot things in the source surface area)